src/web/jcc/client/components/RCControllerPanel.tsx

```diff
@@ -86,30 +90,66 @@ export default class RCControllerPanel extends React.Component {
             // clicks on a new bot details window
             botId: 0,
             isMaximized: true,
+            overdriveEnabled: false,
         };
     }
 
     updateThrottleDirectionMove(event: IJoystickUpdateEvent) {
-        let bin: { binNumber: number; binValue: number } = { binNumber: 0, binValue: 0 };
+        let throttleBin: bin = { binNumber: 0, binValue: 0 };
         if (event.direction.toString() === "FORWARD") {
-            this.calcThrottleBinNum(event.y * 100, "FORWARD", bin);
-            this.props.remoteControlValues.pid_control.throttle = bin.binValue;
+            this.calcThrottleBinNum(event.y * 100, "FORWARD", throttleBin);
+            this.props.remoteControlValues.pid_control.throttle = throttleBin.binValue;
         } else if (event.direction.toString() === "BACKWARD") {
-            this.calcThrottleBinNum(event.y * 100, "BACKWARD", bin);
-            this.props.remoteControlValues.pid_control.throttle = bin.binValue;
+            this.calcThrottleBinNum(event.y * 100, "BACKWARD", throttleBin);
+            this.props.remoteControlValues.pid_control.throttle = throttleBin.binValue;
         }
         this.setState(
-            { throttleDirection: event.direction.toString(), throttleBinNumber: bin.binNumber },
+            { throttleDirection: event.direction.toString(), throttleBinNumber: throttleBin.binNumber },
             () => {},
         );
     }
+    
+    /**
+     * Checks to see if Overdrive has been enabled
+     *
+     * @returns {boolean}
+     */
+
+    isOverdriveChecked() {
+        return this.state.overdriveEnabled;
+    }
+
+    /**
+     * Posts a confirmation dialog to confirm use of Overdrive
+     * If confirmed enables Overdrive
+     *
+     * @returns {void}
+     */
+    
+    async handleOverdriveCheck() {
+
+        if (!this.state.overdriveEnabled){
+            if (
+                !(await CustomAlert.confirmAsync(
+                    "You are about to enable Overdrive.  \nUse Overdrive with caution as it can make the bots difficult to control",
+                    "Enable Overdrive"
+                ))
+            ) {
+                return;
+            }  
+        }
+
+        this.state.overdriveEnabled = !this.state.overdriveEnabled;
```

Comment:
Need to use the setState method here:
`this.setState({overdriveEnabled: !this.state.overdriveEnabled});`